vm-ubuntu-2204-amd64-uefi/gh-actions-overlay.nix

```diff
@@ -0,0 +1,144 @@
+{
+  pkgs ? import <nixpkgs> {},
+  system ? builtins.currentSystem,
+  image ? pkgs.callPackage ./default.nix {},
+  ...
+}: let
+  inherit (pkgs) callPackage dasel;
+  githubRunner = import ./github-runner.nix {inherit pkgs;};
+  overlayedImage = pkgs.vmTools.runInLinuxVM (
+    pkgs.runCommand "nixos-sun-baseline-image"
+    {
+      memSize = 768;
+      preVM = ''
+        mkdir -p $out
+
+        # Locate the image manifest within the store expression:
+        IMAGE_HASH="$(cat "${image}/image.txt")"
+        MANIFEST_PATH="${image}/images/''${IMAGE_HASH:0:2}/''${IMAGE_HASH:2:2}/''${IMAGE_HASH:4:2}/$IMAGE_HASH"
+
+        # Extract the name of the "head" blob in the image:
+        HEAD_BLOB="$(${dasel}/bin/dasel -f "$MANIFEST_PATH" -r toml -w - '.org\.tockos\.treadmill\.manifest-ext\.base\.attrs.org\.tockos\.treadmill\.image\.qemu_layered_v0\.head')"
+
+        # Retrieve the head blob's SHA-256 digest:
+        BLOB_HASH="$(${dasel}/bin/dasel -f "$MANIFEST_PATH" -r toml -w - '.org\.tockos\.treadmill\.manifest-ext\.base\.blobs.'"$HEAD_BLOB"'.org\.tockos\.treadmill\.manifest-ext\.base\.sha256-digest')"
+        BLOB_PATH="${image}/blobs/''${BLOB_HASH:0:2}/''${BLOB_HASH:2:2}/''${BLOB_HASH:4:2}/$BLOB_HASH"
+
+        # The below variable is magically picked up and exposed to the VM.
+        diskImage=disk.qcow2
+        echo "Creating $diskImage based on $BLOB_PATH"
+        ${pkgs.qemu}/bin/qemu-img create -b "$BLOB_PATH" -F qcow2 -f qcow2 "$diskImage"
+      '';
+      postVM = ''
+        mkdir -p $out
+        cp $diskImage $out/overlay.qcow2
+      '';
+      buildInputs = [];
+      passthru.backingImage = image;
+    }
+    ''
+      mkdir -p /mnt
+      ${pkgs.mount}/bin/mount /dev/vda2 /mnt
+      ls /mnt
+
+      cp -r ${githubRunner} /mnt/opt/gh-actions-runner
+
+      cat > "/mnt/etc/systemd/system/gh-actions-runner.service" <<EOF
+      [Unit]
+      Description=GitHub Actions Runner
+      After=network.target tml-puppet.service
+      Requires=tml-puppet.service
+
+      [Service]
+      ExecStartPre=/bin/bash -c 'export REPO_URL=\$(cat /run/tml/parameters/gh-actions-runner-repo-url) && export TOKEN=\$(cat /run/tml/parameters/gh-actions-runner-token) && export JOB_ID=\$(cat /run/tml/job-id) && /opt/gh-actions-runner/config.sh --url \$REPO_URL --token \$TOKEN --name tml-ghactionsrunner-\$JOB_ID --labels tml-ghactionsrunner-\$JOB_ID --unattended --ephemeral'
```

Comment:
Systemd says something about not finding an interpreter in the Nix store for the `config.sh` script. Can we transition this whole expression away from `mkDerivation` to just a plan old `derivation`? The Nix magic causes more troubles than is worth here, I think.

(Compare the "treadmill-store" derivation object in `default.nix`)